txpool/tx_pool_test.go

```diff
@@ -411,3 +417,168 @@ func TestBeforeVIP191Add(t *testing.T) {
 
 	assert.Equal(t, "tx rejected: unsupported features", err.Error())
 }
+
+func TestPoolLimit(t *testing.T) {
+	// synced
+	pool := newPoolWithParams(2, 1, "", "", uint64(time.Now().Unix()))
+	defer pool.Close()
+
+	trx1 := newTx(pool.repo.ChainTag(), nil, 21000, tx.BlockRef{}, 100, nil, tx.Features(0), devAccounts[0])
+	trx2 := newTx(pool.repo.ChainTag(), nil, 21000, tx.BlockRef{}, 100, nil, tx.Features(0), devAccounts[0])
+	pool.add(trx1, false, false)
+
+	err := pool.add(trx2, false, false)
+	assert.Equal(t, "tx rejected: account quota exceeded", err.Error())
+
+	// not synced
+	pool = newPool(2, 1)
```

Comment:
is it worth to split it to 2 different tests ? for sync and not sync versions ?

Reply:
In the code, there are different branches around chain synced and unsynced.

Reply:
gotcha, as long as it is fine with you guys, just thought maybe 2 tc would be more visible to reader 👍